pkgs/by-name/th/the-dark-mod/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  symlinkJoin,
+  writeShellApplication,
+  makeDesktopItem,
+  iconConvTools,
+
+  addDriverRunpath,
+  libGL,
+  xorg,
+  alsa-lib,
+
+  the-dark-mod-unwrapped,
+  the-dark-mod-assets,
+}:
+let
+  libPath = lib.makeLibraryPath [
+    # GLFW
+    addDriverRunpath.driverLink
+    libGL
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXext
+    xorg.libXrandr
+    xorg.libXxf86vm
+
+    # OpenAL
+    alsa-lib
+  ];
+
+  runScript = writeShellApplication {
+    name = "the-dark-mod";
+    runtimeInputs = [ xorg.lndir ];
+
+    text = ''
+      gameDir=''${XDG_DATA_HOME:-$HOME/.local/share}/darkmod
+      mkdir -p "$gameDir"
+      lndir -silent ${the-dark-mod-assets} "$gameDir"
+
+      # Log directory override - by default TDM puts the logs directly in PWD.
+      # We don't want that
+      logDir=''${XDG_STATE_HOME:-$HOME/.local/state}/darkmod
+      mkdir -p "$logDir"
+      rm "$gameDir/darkmod.ini"
+      cat >"$gameDir/darkmod.ini" <<EOF
+      [Debug]
+      LogFile=$logDir/Darkmod.log
+      EOF
+
+      export LD_LIBRARY_PATH=${libPath}:$LD_LIBRARY_PATH
+
+      # I don't know why openal-soft doesn't detect this by itself, but without
+      # this there is no audio :p
+      export ALSOFT_CONF=$gameDir/alsoft.ini
+
+      ${lib.getExe the-dark-mod-unwrapped} \
+        +set fs_basepath "$gameDir" \
```

Comment:
I assume this is for the assets? If so, why can't it just directly point to the nix store and instead need linking into `XDG_DATA_HOME`?

Reply:
It really wants to write things directly next to itself, or to the game data folder. Internal comments indicate that `fs_gamepath` is *supposed* to be read-only but that's just not the case after very extensive testing

(I've genuinely wasted ~3-4 hours trying to coax it into working. It just doesn't.)

Reply:
Would upstream be receptive to a bug report (definitely not a blocker for merging the package)?

Reply:
Certainly